lib/src/common/cache/cache.dart

```diff
@@ -327,23 +372,35 @@ class Cache<TIdentifier, TValue> extends Object with Disposable {
     return new Future.value();
   }
 
-  /// Run [callback] on [TValue] associated with [id].
+  /// Returns `true` and calls [callback] if there is a cached, unreleased
+  /// [TValue] associated with [id]; returns `false` otherwise.
   ///
-  /// If [TIdentifier] [TValue] pair found runs [callback] and returns true,
-  /// otherwise returns false. Will not run on released items and does not
-  /// perform a [get] or [getAsync]. As [get] or [getAsync] are not performed
-  /// calling [applyToItem] will not affect retention or removal of a
-  /// [TIdentifier][TValue] pair from the cache.
+  /// This does not perform a [get] or [getAsync], and as a result, will not
+  /// affect retention or removal of a [TIdentifier][TValue] pair from the cache.
+  /// If any callbacks are in fight on removal of [TIdentifier] [TValue] pair
+  /// [didRemove] will not event until callback has completed.
```

Comment:
This could probably be made clearer and its own paragraph.

```dart
///
/// Any [TIdentifier] [TValue] pair removals will wait for the Future returned by
/// the call to [callback] before emitting [didRemove] events.
```

Also, perhaps I missed this somewhere, but why is this the desired behavior?

Reply:
https://github.com/Workiva/w_common/pull/90#issuecomment-387116341